Classes/UpgradeWizard/SingleToSeperateMmTableUpdater.php

```diff
@@ -0,0 +1,258 @@
+            ->getConnectionForTable(self::TABLE_OLD_MM);
+        try {
+            while ($recordToUpdate = $statement->fetchAssociative()) {
+                try {
+                    $connection->beginTransaction();
+                    $this->migrateRecord($recordToUpdate);
+
+                    // Double-check insertion then delete the record from old table
+                    if ($this->isRecordMigrated($recordToUpdate)) {
+                        $connection->delete(
+                            $this->getOldTableName(),
+                            [
+                                'fieldname' => $recordToUpdate['fieldname'],
+                                'uid_foreign' => $recordToUpdate['uid_foreign'],
+                                'uid_local' => $recordToUpdate['uid_local'],
+                            ],
+                        );
+                    }
+                    $connection->commit();
+                } catch (\Exception $exception) {
+                    $connection->rollBack(); // Revert changes on failure
+                    $this->logError($exception);
+                }
+            }
+        } catch (\Exception $exception) {
+            $this->logError($exception);
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @param array<string, mixed> $recordToUpdate
+     */
+    private function isRecordMigrated(array $recordToUpdate): bool
+    {
+        try {
+            // Determine the table based on the fieldname
+            $table = $recordToUpdate['fieldname'] === self::OLD_DEPARTMENT_FIELD_NAME
+                ? self::TABLE_NEW_DEPARTMENT_MM
+                : self::TABLE_NEW_SUBJECT_FIELD_MM;
+
+            // Check if the record was successfully migrated to the new table
+            $queryBuilder = $this->getConnectionPool()
+                ->getQueryBuilderForTable($table);
+            // Query to check if the record exists
+            $recordExists = $queryBuilder
+                ->count('*')
+                ->from($table)
+                ->where(
+                    $queryBuilder->expr()->eq(
+                        'uid_local',
+                        $queryBuilder->createNamedParameter($recordToUpdate['uid_local'], Connection::PARAM_INT),
+                    ),
+                    $queryBuilder->expr()->eq(
+                        'uid_foreign',
+                        $queryBuilder->createNamedParameter($recordToUpdate['uid_foreign'], Connection::PARAM_INT),
+                    ),
+                )
+                ->executeQuery()
+                ->fetchOne();
+
+            return (bool)$recordExists;
+        } catch (\Exception $exception) {
+            $this->logError($exception);
+
+            return false;
+        }
+    }
+
+    public function getPrerequisites(): array
+    {
+        return [
+            DatabaseUpdatedPrerequisite::class,
+        ];
+    }
+
+    protected function getOldTableName(): string
```

Comment:
Why you need a special method for that? Using the const directly is absolutely OK for me. Method can be removed.